WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -26,10 +26,76 @@ onPage('competitions', function() {
     }).trigger("change");
   }
 
+});
+
+
+// Initializes the competitions map and marker cluster.
+function initializeCompetitionsMap() {
+  competitionsMap = new google.maps.Map(document.getElementById('competitions-map'), {
+    zoom: 2,
+    center: {lat: 0, lng: 0},
+    scrollwheel: true
+  });
+
+  competitionsMarkerCluster = new MarkerClusterer(competitionsMap, [], {
+    maxZoom: 10,
+    clusterSize: 30
+  });
+}
+
+// Sets map container height.
+function resizeMapContainer() {
+  var formHeight = $('#comp-query-form').outerHeight(true);
+  var footerHeight = $('.footer').outerHeight(true);
+  var viewHeight = $(window).innerHeight();
+  var mapHeight = viewHeight - footerHeight - formHeight;
+
+  if(mapHeight < 300) {
```

Comment:
I prefer to write these sorts of boundary conditions as `mapHeight = Math.min(300, mapHeight)`